contrib/analysis/generate-pbench-timeseries-graphs

```diff
@@ -58,6 +58,17 @@ parser.add_argument('-n', "--dont-generate-graphs", action='store_true', help="D
 parser.add_argument('files', metavar='file', type=str, nargs="*", help="files")
 args = parser.parse_args()
 
+match_res = []
+nomatch_re = None
+initialized_matches = False
+if args.match:
+    for pattern in args.match:
+        match_res.append(re.compile(pattern))
+if args.dont_match:
+    nomatch = "|"
+    nomatch = "(%s)" % nomatch.join(args.dont_match)
+    nomatch_re = re.compile(nomatch)
```

Comment:
Just out of curiosity, why are we using different idioms for the positive and negative matches?

Reply:
The intended use case here is to easily process selected data from selected iterations, i. e. constraints from different levels of the tree.  For example, if I want to process pidstat data from iteration 10, I can use

```
-m /10- -m /pidstat
```
If we use the union of all matches, it becomes very awkward to express something like this.  It's a bit awkward if we want to look at iterations 10 and 11, but it's not too bad.  For example, if I want to process pidstat and proc-interrupt data from iterations 10 and 11, it becomes

```
-m '/(10|11)-' -m '/(pidstat|proc-interrupt)'
```
If I want to filter out certain patterns, such as per-CPU, per-IRQ, or per-PID data, then I could use that syntax, or I could use multiple non-match expressions.  But it's not clear to me what the use case for "don't process filenames matching all --dont-match expressions" would be.

```
-m /(10|11)-' -m '(pidstat|proc-interrupt)' -M /pids/ -M by-irq_IRQ
```

Reply:
I concur that generalized match/search syntax is a challenge.  The more comprehensive CLI implementations that I've seen apply the match/don't-match filters in sequence (e.g., a sequence of `-m`'s would be a conjunction, subsequent `-M`'s would filter the initial result, and then subsequent `-m`'s would form a conjunction and select from the filtered result, and so on), but that seems like an awful lot of work, and I'm not sure that it covers all cases.

Regardless, what I was actually asking here was, why are matches represented as a list of individual REs, while don't-matches are combined into a single RE?  (Is not what is good for the goose sauce for the gander?)